bookstore/spiders/books.py

```diff
@@ -0,0 +1,48 @@
+import scrapy
+from scrapy.http import Response
+from bookstore.items import Book
+
+
+class BooksSpider(scrapy.Spider):
+    name = "books"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/"]
+
+    def parse(self, response: Response):
+        book_detail_links = response.css(".product_pod > h3 > a::attr(href)").getall()
+
+        for link in book_detail_links:
+            yield response.follow(link, callback=self.parse_book)
+
+        next_page = response.css(".next > a::attr(href)").get()
+        if next_page is not None:
+            yield response.follow(next_page, callback=self.parse)
+
+    @staticmethod
+    def _get_amount_in_stock(response: Response):
+        stock_text = response.css("table.table td::text").getall()
+        print("Stock text:", stock_text)
+        if len(stock_text) >= 6:
+            stock_quantity_text = stock_text[5]
+            stock_quantity = "".join(filter(str.isdigit, stock_quantity_text))
+            print("Stock quantity:", stock_quantity)
+            if stock_quantity:
+                return int(stock_quantity)
+        return 0
+
+    @staticmethod
+    def _get_rating(response: Response):
+        rating_classes = response.css("p.star-rating::attr(class)").get()
+        rating_map = {"One": 1, "Two": 2, "Three": 3, "Four": 4, "Five": 5}

```

Comment:
let's move it to constant at the top of the file, so that it'll be easier to find and change ratings :)

Reply:
Fixed